pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -449,7 +439,33 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
         return outcome_map;
     }
 
+    void
+    setDeviceSeed(const std::optional<std::size_t> &deviceSeed = std::nullopt) {
+        this->DeviceSeed = deviceSeed;
+    }
+
+    void setDeviceSeedFromPRNG(std::mt19937 *gen) {
```

Comment:
I am still a bit uncomfortable with having two seed setter suites. As far as I can tell, the purpose of this second suite is:
- A measurement, say `LightningSimulator::Probs()` in catalyst, would call this `setDeviceSeedFromPRNG` on the measurement class. This adds a seed number to the measurement class.
- When the measurement class actually performs the measurements (e.g. in generate sameples), we `activateDeviceSeed` to produce the seeded innate prng object of the measurement class.

I still think this logic can be performed inside the two existing setters. The goal of the entire pipeline here is step 2, aka perform `rng.seed(seed)`. I think the catalyst measurements can just pass the seed number into the measurement class without passing the catalyst device prng object. This way we can just use the old `setSeed(size_t)` methods. In fact, there only needs to be one setseed method that handles everything:
```
void setSeed(optional int seed){
   if seed:
      rng.seed(seed);
      this.DeviceSeed = seed;
   else:
      rng.seed(rd());
}
``` 

And it can handle the various callers, whether giving a seed or not.

I'm fine with the measurement base saving the seed number for utility, e.g. using the seed number it saw for LK's special `generate_samples`.

What do you think? I think this would be a lot cleaner.

Reply:
Done